docker/linux/Dockerfile

```diff
@@ -0,0 +1,151 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```

Comment:
How about we change docker/linux/Dockerfile to docker/linux? The folder docker already declare that this is place to put Dockfile.

Reply:
Changing the name from `Dockerfile` is actually kind of annoying for some clients like `buildctl` which will actually build this faster, also it breaks syntax highlighting since there is no known extension, just a filename standard.

It also gives us a folder for OS specific assets that might want to be included which might be nice.

Reply:
Ok, understand.